src/listfunc.c

```diff
@@ -225,9 +225,16 @@ static Obj FuncAPPEND_LIST_INTR(Obj self, Obj list1, Obj list2)
     RequireSmallList(SELF_NAME, list2);
 
     /* handle the case of strings now */
-    if (IS_STRING_REP(list1) && IS_STRING_REP(list2)) {
-        AppendString(list1, list2);
+    if (IS_STRING_REP(list1)) {
+      if(IS_STRING(list2)) {
+        if(IS_STRING_REP(list2)) {
+          AppendString(list1, list2);
+        }
+        else {
+          AppendString(list1, ImmutableString(list2));
+        }
         return 0;
+      }
     }
```

Comment:
Code formatting!

How about simplifying to this:
```suggestion
    if (IS_STRING_REP(list1) && IS_STRING(list2)) {
        if (!IS_STRING_REP(list2)) {
            list2 = ImmutableString(list2);
        }
        AppendString(list1, list2);
        return 0;
    }
```

Of course even simpler would be this, but yeah, we kinda wanted to get away from automatic in-place conversions, so yeah...
```suggestion
    if (IS_STRING_REP(list1) && IsStringConv(list2)) {
        AppendString(list1, list2);
        return 0;
    }
```